rust/src/dataset.rs

```diff
@@ -381,16 +379,27 @@ impl Dataset {
                     }
                 }
 
-                let builder = IvfPqIndexBuilder::try_new(
+                let ivf_params = IvfBuildParams {
+                    num_partitions: vec_params.num_partitions as usize,
+                    metric_type: vec_params.metric_type,
+                    max_iters: 50,
+                };
+                let pq_params = PQBuildParams {
+                    num_sub_vectors: vec_params.num_sub_vectors as usize,
+                    num_bits: 8,
```

Comment:
do we have to change much to support configurable num_bits? is it just exposing a new API parameter? or is the underlying index creation hard coded to 8 bits?

Reply:
`num_nbits` is not configurable via user-facing API yet.  8 bits is just 1 byte. other than `8 bits`, the other configuration needs some work .